### PR TITLE
Show medical history in summary

### PR DESCRIPTION
**Pull request by saket**
Huge PR because migration

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/storage/Migration_13_14.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.threeten.bp.Instant
+import java.util.UUID
+
+/**
+ * Adds an empty [MedicalHistory] for all patients.
+ */
+@Suppress("ClassName")
+class Migration_13_14 : Migration(13, 14) {
+
+  override fun migrate(db: SupportSQLiteDatabase) {
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/storage/Migration_13_14.kt]: This migration needs to be thoroughly reviewed and tested. Steps to test:

1. Check if there is a patient present on staging that *doesn't* have a medical history stored. This is not the same as a medical history with `None of the above` as the selected answer. If you cannot find any, manually insert one by either running `PatientSyncAndroidTest` or checking out a version of master before medical history was introduced.

2. Run this branch.

3. Open the patient's summary screen

4. The screen shouldn't crash. Scroll to the bottom to find medical history. The history should be empty with no selected answers.

File: app/src/main/java/org/simple/clinic/storage/Migration_13_14.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    db.query("SELECT * FROM `Patient`").use {
+      val nowTime = Instant.now()
+      val syncStatus = "PENDING"
+      val falseAsInt = 0
+
+      while (it.moveToNext()) {
+        val patientUuid = it.getString(it.getColumnIndex("uuid"))
+        val historyUuid = UUID.randomUUID()
+        db.execSQL("""
+          INSERT INTO `MedicalHistory` VALUES(
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/storage/Migration_13_14.kt]: TODO: Clear the table before inserting.

File: app/src/main/java/org/simple/clinic/summary/SummaryMedicalHistoryItem.kt [hunk 3]
```diff
@@ -35,29 +36,26 @@ data class SummaryMedicalHistoryItem(
...
-    holder.strokeQuestionView.apply {
-      render(HAS_HAD_A_STROKE)
-      checkBox.isChecked = medicalHistory.hasHadStroke
-    }
-
-    holder.kidneyDiseaseQuestionView.apply {
-      render(HAS_HAD_A_KIDNEY_DISEASE)
-      checkBox.isChecked = medicalHistory.hasHadKidneyDisease
-    }
+    val questionViewToQuestions = mapOf(
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/summary/SummaryMedicalHistoryItem.kt]: TODO: Move it later.

File: app/src/main/java/org/simple/clinic/medicalhistory/MedicalHistoryRepository.kt [hunk 4]
```diff
@@ -14,6 +16,18 @@ class MedicalHistoryRepository @Inject constructor(
     private val dao: MedicalHistory.RoomDao
 ) : SynceableRepository<MedicalHistory, MedicalHistoryPayload> {
 
+  // TODO: Android Test.
+  fun historyForPatient(patientUuid: PatientUuid): Observable<MedicalHistory> {
+    return dao.historyForPatient(patientUuid)
+        .map { patients ->
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/medicalhistory/MedicalHistoryRepository.kt]: TODO: In a later PR.

File: app/src/main/java/org/simple/clinic/summary/PatientSummaryScreenController.kt [hunk 5]
```diff
@@ -94,7 +94,10 @@ class PatientSummaryScreenController @Inject constructor(
 
     val medicalHistoryItems = patientUuid
         .flatMap { medicalHistoryRepository.historyForPatient(it) }
-        .map(::SummaryMedicalHistoryItem)
+        .map { history ->
+          val lastSyncTimestamp = timestampGenerator.generate(history.updatedAt)
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/summary/PatientSummaryScreenController.kt]: TODO: In a later PR.

**Pull request merged**